extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -427,34 +450,54 @@ private static void extractClientVersionAndKey() throws IOException, ExtractionE
     }
 
     /**
-     * Get the client version
+     * Get the client version used by YouTube website on InnerTube requests.
      */
     public static String getClientVersion() throws IOException, ExtractionException {
-        if (!isNullOrEmpty(clientVersion)) return clientVersion;
-        if (areHardcodedClientVersionAndKeyValid()) {
-            return clientVersion = HARDCODED_CLIENT_VERSION;
+        if (!isNullOrEmpty(clientVersion)) {
+            return clientVersion;
         }
 
         extractClientVersionAndKey();
```

Comment:
What? Why do you always extract client version and key even if the hardcoded ones are ok? Also, why did you make this change here, even though it has nothing to do with adding support of other delivery methods? (same below in `getKey()`

Reply:
It is to prevent fingerprinting basing on client version but like you said (and for the clients version update), this should not be changed with this PR. I will revert both things I mentioned.

Reply:
Done